WalletWasabi/Blockchain/BlockFilters/FilterLinkedList.cs

```diff
@@ -0,0 +1,87 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using WalletWasabi.Backend.Models;
+
+namespace WalletWasabi.Blockchain.BlockFilters;
+
+public class FilterLinkedList : IEnumerable<FilterModel>
+{
+	record Node(FilterModel Filter, Node? Next)
+	{
+		public int Count { get; } = (Next?.Count ?? 0) + 1;
```

Comment:
I've never used a construction like this before. I was wondering... So this is a chain of calls down to the very last filter every time when you call `Count()`?

Reply:
No. It is not a chain or call and it is not similar to call `.Count` in `Enumerable`. It is easy to see that this is a property being assigned at construction time and not at calculated recursive thing.